src/Countries/Malaysia.php

```diff
@@ -0,0 +1,535 @@
+        return 'my';
+    }
+
+    protected function allHolidays(int $year): array
+    {
+        return array_merge(
+            $this->fixHolidays($year),
+            $this->variableHolidays($year)
+        );
+    }
+
+    /**
+     * @param int $year
+     * @return array<string, CarbonImmutable>
+     */
+    protected function fixHolidays(int $year): array
+    {
+        return [
+            'Hari Pekerja' => CarbonImmutable::createFromDate($year, 5, 1, $this->timezone),
+            'Hari Kebangsaan' => CarbonImmutable::createFromDate($year, 8, 31, $this->timezone),
+            'Hari Malaysia' => CarbonImmutable::createFromDate($year, 9, 16, $this->timezone),
+            'Hari Krismas' => CarbonImmutable::createFromDate($year, 12, 25, $this->timezone),
+        ];
+    }
+
+    /**
+     * @param int $year
+     * @return array<string, CarbonImmutable|string>
+     */
+    protected function variableHolidays(int $year): array
+    {
+        $variableHolidays = [
+            'Tahun Baru Cina' => $this->chineseCalendar('01-01', $year),
+            'Tahun Baru Cina Hari Kedua' => $this->chineseCalendar('01-02', $year),
+            'Hari Raya Aidilfitri' => $this->islamicCalendar('01/10', $year),
+            'Hari Raya Aidilfitri Hari Kedua' => $this->islamicCalendar('02/10', $year),
+            'Hari Wesak' => $this->hariWesak($year),
+            'Hari Raya Aidiladha' => $this->hariAidiladha($year),
+            'Awal Muharram' => $this->awalMuharram($year),
+            'Maulidur Rasul' => $this->maulidurRasul($year),
+        ];
+
+        $regionHolidays = $this->holidaysByRegion($year);
+
+        if ($this->hariKeputeraanYDP($year)) {
+            $variableHolidays['Hari Keputeraan Yang Di-Pertuan Agong'] = $this->hariKeputeraanYDP($year);
+        }
+
+        return array_merge($variableHolidays, $regionHolidays);
+    }
+
+    /**
+     * return holiday by region, if not set return all
+     * 
+     * @param int $year
+     * @return array<string, CarbonImmutable|string>
+     */
+    protected function holidaysByRegion(int $year): array
+    {
+        return match ($this->region) {
+            'jhr' => $this->regionJohor($year),
+            'kdh' => $this->regionKedah($year),
+            'ktn' => $this->regionKelantan($year),
+            'kul' => $this->regionKualaLumpur($year),
+            'lbn' => $this->regionLabuan($year),
+            'mlk' => $this->regionMelaka($year),
+            'nsn' => $this->regionNegeri9($year),
+            'phg' => $this->regionPahang($year),
+            'png' => $this->regionPenang($year),
+            'prk' => $this->regionPerak($year),
+            'pls' => $this->regionPerlis($year),
+            'pjy' => $this->regionPutrajaya($year),
+            'sbh' => $this->regionSabah($year),
+            'swk' => $this->regionSarawak($year),
+            'sgr' => $this->regionSelangor($year),
+            'trg' => $this->regionTerengganu($year),
+            default => array_merge(
```

Comment:
By default I wouldn't return any region specific holidays. What you think?

Reply:
i think it is better to add check if region is set; then throw a new exception invalid region if no match. better flow right? @Nielsvanpach